install.sh

```diff
@@ -0,0 +1,276 @@
+      INSTALL_PATH="$2"
+      shift 2
+      ;;
+    -h|--help)
+      echo "Usage: install.sh [options]"
+      echo "Options:"
+      echo "  -v, --version VERSION    Install specific version"
+      echo "  -d, --dir DIRECTORY      Install to specific directory"
+      echo "  -h, --help               Show this help message"
+      exit 0
+      ;;
+    *)
+      warn "Unknown option: $1"
+      shift
+      ;;
+  esac
+done
+
+if [[ ! -d "$INSTALL_PATH" ]]; then
+  ohai "Creating install directory: $INSTALL_PATH"
+  mkdir -p "$INSTALL_PATH" || abort "Failed to create directory: $INSTALL_PATH"
+fi
+
+if [[ ! -w "$INSTALL_PATH" ]]; then
+  abort "No write permission to $INSTALL_PATH. Try running with sudo or specify a different directory with --dir."
+fi
+
+TMP_DIR="$(mktemp -d)"
+cleanup() {
+  rm -rf "$TMP_DIR"
+}
+trap cleanup EXIT
+
+if [[ "$VERSION" == "latest" ]]; then
+  ohai "Fetching latest release information..."
+  RELEASE_URL="https://api.github.com/repos/agentuity/cli/releases/latest"
+  VERSION=$(curl -s $RELEASE_URL | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+  if [[ -z "$VERSION" ]]; then
+    abort "Failed to fetch latest version information"
+  fi
+fi
+
+VERSION="${VERSION#v}"
+
+if [[ "$OS" == "Windows" ]]; then
+  if [[ "$ARCH" == "x86_64" ]]; then
+    DOWNLOAD_FILENAME="agentuity-x64.msi"
+  elif [[ "$ARCH" == "arm64" ]]; then
+    DOWNLOAD_FILENAME="agentuity-arm64.msi"
+  else
+    DOWNLOAD_FILENAME="agentuity-x86.msi"
+  fi
+else
+  DOWNLOAD_FILENAME="agentuity_${OS}_${ARCH}.${EXTENSION}"
+fi
+
+DOWNLOAD_URL="https://github.com/agentuity/cli/releases/download/v${VERSION}/${DOWNLOAD_FILENAME}"
+
+ohai "Downloading Agentuity CLI v${VERSION} for ${OS}/${ARCH}..."
+curl -L --progress-bar "$DOWNLOAD_URL" -o "$TMP_DIR/$DOWNLOAD_FILENAME" || abort "Failed to download from $DOWNLOAD_URL"
+
+ohai "Processing download..."
+if [[ "$OS" == "Windows" ]]; then
+  ohai "Downloaded Windows MSI installer to $TMP_DIR/$DOWNLOAD_FILENAME"
+  ohai "To install, run the MSI installer manually"
+  echo "  $TMP_DIR/$DOWNLOAD_FILENAME"
+  cp "$TMP_DIR/$DOWNLOAD_FILENAME" "$HOME/" || abort "Failed to copy MSI to $HOME/"
+  ohai "MSI installer copied to $HOME/$DOWNLOAD_FILENAME"
+  exit 0
+elif [[ "$EXTENSION" == "tar.gz" ]]; then
+  ohai "Extracting..."
+  tar -xzf "$TMP_DIR/$DOWNLOAD_FILENAME" -C "$TMP_DIR" || abort "Failed to extract archive"
+elif [[ "$EXTENSION" == "zip" ]]; then
+  ohai "Extracting..."
+  unzip -q "$TMP_DIR/$DOWNLOAD_FILENAME" -d "$TMP_DIR" || abort "Failed to extract archive"
+else
+  abort "Unknown archive format: $EXTENSION"
+fi
+
+ohai "Installing to $INSTALL_PATH..."
+if [[ -f "$TMP_DIR/agentuity" ]]; then
+  cp "$TMP_DIR/agentuity" "$INSTALL_PATH/" || abort "Failed to copy binary to $INSTALL_PATH"
+  chmod +x "$INSTALL_PATH/agentuity" || abort "Failed to make binary executable"
+else
+  abort "Binary not found in extracted archive"
+fi
+
+if command -v "$INSTALL_PATH/agentuity" >/dev/null 2>&1; then
```

Comment:
can we also test with output of the version command and compare to version downloaded?